### ISSUE TITLE
Uncaught exception from 'path' when no main is provided in package.json

### ISSUE THREAD

**Issue by pamtbaau** [yan-foto/electron-reload#37]
When `package.json` does not contain a `main` property, `path` will throw an uncaught exception at line 15 of your main.js.

Would be handy if it is mentioned in the readme, and/or if the exception from path is caught and rethrow with a more indicative message.

**Comment by yan-foto** [comment]
I think it was in one of the PRs that the main file was added to chokidar ignore files... However, your're right! the main file is only required for hard resets. On the other hand, I cannot imagine how an electron app is to run without the [main process](https://github.com/electron/electron/blob/master/docs/tutorial/quick-start.md#main-process) 🤔

**Comment by pamtbaau** [comment]
I have in my package.json
```
  "scripts": {
    "start": "electron .",
    ...
  },
```
Starting app from command prompt using `npm start` or using key-shortcut in vscode to run a task running  the same command works fine.

**Comment by yan-foto** [comment]
I don't get it 🤔 Your electron app does have a main file or not? If not, how is your app running in electron in the first place?

**Comment by pamtbaau** [comment]
Maybe the  `"start": "electron ."` might have confused you...

When starting electron with `electron .`, Electron will search for a local `index.js` file which becomes the `main` process. See **Note:** in [docs](https://electron.atom.io/docs/tutorial/quick-start/#write-your-first-electron-app). 

`electron .` behaves differently from `electron index.js` though. When pointing to the main file (`index.js` in my case), every Electron app on your PC will store all meta data into the same user folder called Electron (on windows: `C:\Users\me\AppData\Roaming\Electron`), while pointing to `.` it wil generate a specific user data folder per Electron app using the `"name"` property of the `package.json`  (e.g. on windows: `C:\Users\me\AppData\Roaming\my-app-name`)

I'm looking more into your main.js and different electron startup scenarios and seeings some peculiarities... More about that later.

**Comment by yan-foto** [comment]
Aaaah! Got it. It may be ironic, but it's been a long time since I've used electron. Maybe Its a good idea to look either for explicit main file or implicit `index.js` for the main process! 👍

**Comment by pamtbaau** [comment]
I've done a few tests ( [see READ.ME in repo](https://github.com/pamtbaau/electron-reload-test)) and the following is a solution for my use-case. And only tested for my use-case...

```
  const appWorkingDir = process.cwd();
  const appArgv = process.argv;

  // NB.
  // Process.argv will always have at least 1 argument.
  // If Electron is started without any argument, Electron will start with a 
  // default main and electron-reload will not be called.

  let mainFile;

  if (/\.$/.exec(appArgv[1])) {
    mainFile = path.join(appWorkingDir, 'index.js')
  } else if (/\.js$/.exec(appArgv[1])) {
    mainFile = path.join(appWorkingDir, appArgv[1])
  } else {
    throw new Error('Cannot find file for main process in first argument.');
  }
```

**Comment by yan-foto** [comment]
Fixed in #39
